enterprise/internal/insights/resolvers/dashboard_resolvers_test.go

```
@@ -67,3 +69,50 @@ func TestIsVirtualDashboard(t *testing.T) {
 		t.Fatal()
 	}
 }
+
+func TestHasPermissionForGrants(t *testing.T) {
+	t.Run("No grants returns true", func(t *testing.T) {
+		got := hasPermissionForGrants([]store.DashboardGrant{}, []int{1, 2}, []int{1})
+		if got != true {
+			t.Errorf("should have permission for zero grants")
```

This just got me thinking - is this valid? Do we allow users to create insights or dashboards without any grants? That feels like we are only creating an avenue for abuse / mistakes, since by definition nobody would be able to interact with them? What do you think? (This doesn't have to block this PR, we can raise a new issue)

---

Yeah I guess it doesn't make sense as an actual use case; just trying to cover the function itself I guess. I can add a quick check in to see if there are 0 dashboard grants we can return an error.. I'll add that in now actually, it's a tiny change.

---

LGTM!